lib/tortoise311/connection/inflight.ex

```diff
@@ -244,7 +246,8 @@ defmodule Tortoise311.Connection.Inflight do
       {:keep_state, data, next_actions}
     else
       :error ->
-        {:stop, {:protocol_violation, :unknown_identifier}, data}
+        Logger.warn("[Tortoise311] Unknown identifier in #{inspect(update)}. Ignoring it.")
+        :keep_state_and_data
```

Comment:
If the client is out of sync with the server, shouldn't Tortoise311 close the socket and reconnect? It seems like this should be unrecoverable.

Reply:
I just did a quick breeze of MQTT spec to compare data from the bug report. It looks like the initial data that cause the crash was opcode 4 with identifier 27484. I think that's a PUBACK which seems like something we should actually be handling

I guess I was viewing this PR as getting progress growing but may need to be adapted

Reply:
> I just did a quick breeze of MQTT spec to compare data from the bug report. It looks like the initial data that cause the crash was opcode 4 with identifier 27484. I think that's a PUBACK which seems like something we should actually be handling

PUBACKs are handled. The issue here a PUBACK sent by the MQTT server was aimed at a conversation (a Track) that had vanished. In this case, the Inflight had been reset (or restarted) and all Tracks had vanished.

Reply:
> If the client is out of sync with the server, shouldn't Tortoise311 close the socket and reconnect? It seems like this should be unrecoverable.

I am not entirely sure what is best. Would the MQTT server, upon reconnection, send the same package aimed at a non-existent track, thus causing a repeating reconnect? Ignoring the mis-directed package seems the lesser evil at the moment.